src/frontend/sharedComponents/HookFormTextInput.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import {
+  Control,
+  Controller,
+  FieldValues,
+  Path,
+  useWatch,
+  ValidationRule,
+  useFormState,
+  UseControllerProps,
+} from 'react-hook-form';
+import {TextInput as RNTextInput, StyleSheet, View} from 'react-native';
+import {BLACK, LIGHT_GREY, RED} from '../lib/styles';
+import {ErrorIcon} from './icons';
+import {ViewStyleProp} from '../sharedTypes';
+import {Text} from './Text';
+
+type TextInputProps<InputFields extends FieldValues> = {
+  containerStyle?: ViewStyleProp;
+  showCharacterCount?: boolean;
+  control: Control<InputFields>;
+} & Pick<
+  React.ComponentProps<typeof RNTextInput>,
+  'placeholder' | 'placeholderTextColor'
+> &
+  Omit<UseControllerProps<InputFields>, 'control'>;
```

Comment:
`control` is optional (there is a `formProvider` that allows for control to be passed to useController), so I made it required here